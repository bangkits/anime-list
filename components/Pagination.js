/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { Stack, Pagination } from "@mui/material"

const paginationWrapper = css({
  display: 'flex',
  width: '100%',
  justifyContent: 'center',
  marginBottom: 30,
  marginTop: 30
});

const CustomPagination = (props) => {
  const { count, page, handlePageChange } = props;

  return (
    <div css={paginationWrapper}>
      <Stack spacing={6}>
        <Pagination count={count} page={page} onChange={handlePageChange} />
      </Stack>
    </div>
  )
};

export default CustomPagination;