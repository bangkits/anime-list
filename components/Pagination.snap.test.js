import renderer from 'react-test-renderer';
import Pagination from './Pagination';

describe('Pagination Component Snapshot', () => {

  it('Renders Pagination component with 2 pages', () => {
    const props = {
      page: 1,
      count: 2,
      handlePageChange: jest.fn()
    };

    const tree = renderer
      .create(<Pagination {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders Pagination component with 10 pages', () => {
    const props = {
      page: 1,
      count: 10,
      handlePageChange: jest.fn()
    };

    const tree = renderer
      .create(<Pagination {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});