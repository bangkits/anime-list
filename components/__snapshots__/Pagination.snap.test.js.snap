// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Snapshot Renders Pagination component with 2 pages 1`] = `
<div
  className="css-19v8nu-CustomPagination"
>
  <div
    className="css-14w0e8k-MuiStack-root"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Pagination Component Snapshot Renders Pagination component with 10 pages 1`] = `
<div
  className="css-19v8nu-CustomPagination"
>
  <div
    className="css-14w0e8k-MuiStack-root"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            1
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            2
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            3
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            4
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            5
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <div
            className="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
          >
            …
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 10"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            10
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
